plugins/modules/firewalld_info.py

```diff
@@ -0,0 +1,392 @@
+                            description:
+                              - A list of rich language rule.
+                            returned: success
+                            type: list
+                            sample:
+                              - "rule protocol value=\"icmp\" reject"
+                              - "rule priority=\"32767\" reject"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from distutils.version import StrictVersion
+
+
+try:
+    import dbus
+    HAS_DBUS = True
+except ImportError:
+    HAS_DBUS = False
+
+try:
+    import firewall.client as fw_client
+    import firewall.config as fw_config
+    HAS_FIREWALLD = True
+except ImportError:
+    HAS_FIREWALLD = False
+
+
+def get_version():
+    return fw_config.VERSION
+
+
+def get_active_zones(client):
+    return client.getActiveZones().keys()
+
+
+def get_all_zones(client):
+    return client.getZones()
+
+
+def get_default_zone(client):
+    return client.getDefaultZone()
+
+
+def get_zone_settings(client, zone):
+    return client.getZoneSettings(zone)
+
+
+def get_zone_target(zone_settings):
+    return zone_settings.getTarget()
+
+
+def get_zone_icmp_block_inversion(zone_settings):
+    return zone_settings.getIcmpBlockInversion()
+
+
+def get_zone_interfaces(zone_settings):
+    return zone_settings.getInterfaces()
+
+
+def get_zone_sources(zone_settings):
+    return zone_settings.getSources()
+
+
+def get_zone_services(zone_settings):
+    return zone_settings.getServices()
+
+
+def get_zone_ports(zone_settings):
+    return zone_settings.getPorts()
+
+
+def get_zone_protocols(zone_settings):
+    return zone_settings.getProtocols()
+
+
+# This function suports python-firewall 0.9.0(or later).
+def get_zone_forward(zone_settings):
+    return zone_settings.getForward()
+
+
+def get_zone_masquerade(zone_settings):
+    return zone_settings.getMasquerade()
+
+
+def get_zone_forward_ports(zone_settings):
+    return zone_settings.getForwardPorts()
+
+
+def get_zone_source_ports(zone_settings):
+    return zone_settings.getSourcePorts()
+
+
+def get_zone_icmp_blocks(zone_settings):
+    return zone_settings.getIcmpBlocks()
+
+
+def get_zone_rich_rules(zone_settings):
+    return zone_settings.getRichRules()
+
+
+def main():
+    module_args = dict(
+        active_zones=dict(required=False, type='bool', default=False),
+        zones=dict(required=False, type='list', elements='str'),
+    )
+
+    module = AnsibleModule(
+        argument_spec=module_args,
+        supports_check_mode=True,
+    )
+
+    firewalld_info = dict()
+    result = dict(
+        changed=False,
+        active_zones=module.params['active_zones'],
+        collected_zones=list(),
+        undefined_zones=list(),
+        warnings=list(),
+    )
+
+    # Exit with failure message if requirements modules are not installed.
+    if not HAS_DBUS:
+        module.fail_json(msg='python-dbus is required for the firewalld_info module.')
```

Comment:
```suggestion
		from ansible.module_utils.basic import missing_required_lib
        module.fail_json(msg=missing_required_lib('python-dbus'))
```

Reply:
Changed to use missing_required_lib(). Thank you for your advice!